internal/core/item_rendering.rs

```diff
@@ -487,22 +487,84 @@ pub trait ItemRendererFeatures {
 /// After rendering an item, we cache the geometry and the transform it applies to
 /// children.
 #[derive(Clone)]
-pub struct CachedItemGeometryAndTransform {
-    /// The item's reported geometry, via [`ItemRc::geometry`].
-    pub geometry: LogicalRect,
-    /// The item's transform to apply to children, via [`ItemRc::children_transform`].
-    pub boxed_children_transform: Option<Box<ItemTransform>>,
-}
 
-impl From<(LogicalRect, Option<ItemTransform>)> for CachedItemGeometryAndTransform {
-    fn from((geometry, children_transform): (LogicalRect, Option<ItemTransform>)) -> Self {
-        Self { geometry, boxed_children_transform: children_transform.map(Into::into) }
-    }
+pub enum CachedItemGeometryAndTransform {
+    /// A regular item with a translation
+    RegularItem {
+        /// The item's bounding rect relative to its parent.
+        bounding_rect: LogicalRect,
+        /// The item's geometry relative to its parent.
+        geometry: LogicalRect,
+    },
+    /// An item such as Rotate that defines an additional transformation
+    ItemWithTransform {
+        /// The item's bounding rect relative to its parent.
+        bounding_rect: LogicalRect,
+        /// The transform to apply to children.
+        transform: Box<ItemTransform>,
+        /// The item's geometry relative to its parent.
+        geometry: LogicalRect,
```

Comment:
How about a single `geometry_and_transform: Box<(LogicalRect, ItemTransform)>` 
to keep the bigger variant small?

Reply:
Excellent idea :)